graphql/resolvers.go

```diff
@@ -744,6 +744,29 @@ func (r *taskResolver) PatchNumber(ctx context.Context, obj *restModel.APITask)
 	return &patch.PatchNumber, nil
 }
 
+func (r *taskResolver) BaseCommitDuration(ctx context.Context, at *restModel.APITask) (*restModel.APIDuration, error) {
+	t, err := r.sc.FindTaskById(*at.Id)
+	if err != nil {
+		return nil, ResourceNotFound.Send(ctx, fmt.Sprintf("error finding task by id %s: %s", *at.Id, err.Error()))
+	}
+	baseTask, err := t.FindTaskOnBaseCommit()
+	if err != nil {
+		return nil, InternalServerError.Send(ctx, fmt.Sprintf("Error finding task %s on base commit", *at.Id))
+	}
+	if baseTask == nil {
+		return nil, ResourceNotFound.Send(ctx, fmt.Sprintf("Unable to find task %s on base commit", *at.Id))
+	}
+	if baseTask.TimeTaken == 0 {
+		return nil, nil
+	}
+	apiBaseTask := restModel.APITask{}
```

Comment:
Instead of building the entire model, you can just return NewAPIDuration and pass the timetaken